File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        current_digit_sum = 0
        
        # Step 1: Convert characters to numbers and perform the 1st transformation
        for char in s:
            alphabet_position = ord(char) - ord('a') + 1
            # Extract and sum digits of the alphabet position
            while alphabet_position > 0:
                current_digit_sum += alphabet_position % 10
                alphabet_position //= 10
                
        # Step 2: Perform the remaining k - 1 transformations
        k -= 1
        while k > 0:
            next_digit_sum = 0
            while current_digit_sum > 0:
                next_digit_sum += current_digit_sum % 10
                current_digit_sum //= 10
            current_digit_sum = next_digit_sum
            k -= 1
            
        return current_digit_sum

File: 2076-sum-of-digits-of-string-after-convert/sum-of-digits-of-string-after-convert.js
var getLucky = function (s, k) {
    let currentDigitSum = 0;

    // Step 1: Convert characters to numbers and perform the 1st transformation
    for (let i = 0; i < s.length; i++) {
        let alphabetPosition = s.charCodeAt(i) - 97 + 1; // 97 is ASCII for 'a'
        while (alphabetPosition > 0) {
            currentDigitSum += alphabetPosition % 10;
            alphabetPosition = Math.floor(alphabetPosition / 10);
        }
    }

    // Step 2: Perform the remaining k - 1 transformations
    k--;
    while (k > 0) {
        let nextDigitSum = 0;
        while (currentDigitSum > 0) {
            nextDigitSum += currentDigitSum % 10;
            currentDigitSum = Math.floor(currentDigitSum / 10);
        }
        currentDigitSum = nextDigitSum;
        k--;
    }

    return currentDigitSum;
};